### DISCUSSION TITLE
README query

### DISCUSSION THREAD

**Discussion by kruthikahm2000** [EddieHubCommunity/support#412] [Q&A]
How to built a README most creative and best(from basics to advance)?!

**Comment by eddiejaoude** [comment]
I am writing notes on this at the moment, so I will give a short answer now, and another more full answer later.

My thoughts are, this is the first impression someone gets of your repo, so the first page of the README needs to easily show what the project is about and what it looks like.

- [ ] Screenshots: this will help so much
- [ ] Prerequisite: what is needed to be installed first, for example node v12+
- [ ] Quickstart: how can people get started with the basics
- [ ] Configuration / enviroment variables: what configuration is needed, for example database credentials
- [ ] Directory structure: what files are where and why
- [ ] Deployment: what branches deploy to which environment
- [ ] FAQ / trouble shooting: what common issues do people have and what solutions work

**Comment by BOLT04** [comment]
I agree with all the points made by @eddiejaoude 😃. I'd add architecture diagrams (screenshots or perhaps a link to more docs on this) and a roadmap of the project (in case it fits the use case).
Also, perhaps some trivia or insights on why the project has that name 😄 (or other interesting information). I remember Eddie reviewed this project with a really cool name: https://github.com/red-perfume/red-perfume#why-is-it-called-red-perfume. 

Might be worth checking this out @kruthikahm2000: https://github.com/matiassingers/awesome-readme
